Update Docker images to newest versions

lukaszstolarczuk opened the pull request:
Ubuntu 18.04 -> 19.10
Fedora 28 -> 31

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/pmem/pmdk/4110)
<!-- Reviewable:end -->

marcinslusarz commented on utils/docker/images/Dockerfile.ubuntu-19.10:
```diff
@@ -104,7 +104,7 @@ ENV PACKAGING_DEPS "debhelper \
 ENV CODECOV_DEPS curl
 
 # Coverity
-ENV COVERITY_DEPS ruby gcc-6 g++-6 wget
```
AFAIR Coverity doesn't support gcc>6

marcinslusarz commented on utils/docker/images/Dockerfile.fedora-31:
```diff
@@ -35,7 +35,7 @@
 
 # Pull base image
-FROM fedora:28
+FROM fedora:31
```
we probably shouldn't install ndctl from source then...

lukaszstolarczuk commented:
- I've updated gcc to the newest (v9), but if there's going to be a problem with Coverity I'll fix it with separate image for Coverity (at least that's the current plan)
- I've switched installation of ndctl from packages, but the "install-ndctl.sh" script is left in the repo (in case we need it, e.g. for testing or to use it back in the build)
